e2e/testcafe-devextreme/tests/dataGrid/editing/editingEvents.ts

```diff
@@ -172,3 +174,52 @@ testCases.forEach(({ caseName, expected, onRowRemoving }) => {
     });
   });
 });
+
+// T1250405
+test('DataGrid - Canceled rows are hidden when multiple rows are added in batch mode', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+  const addBtn = dataGrid.getToolbar().getItem();
+  const saveBtn = dataGrid.getToolbar().getItem(1);
+  await t
+    .click(addBtn)
+    .pressKey('1')
+    .click(addBtn)
+    .pressKey('2')
+    .click(saveBtn);
+
+  const overlay1 = new Overlay(Selector('.dx-overlay-wrapper'), 0);
+  const overlay2 = new Overlay(Selector('.dx-overlay-wrapper'), 1);
+  const cancelBtn = overlay2.getToolbar(1).getItem(1);
+  const saveBtnPopup = overlay1.getToolbar(1).getItem(0);
+
+  await t
+    .click(cancelBtn)
+    .click(saveBtnPopup);
+  await t
```

Comment:
Also, I recommend adding this check here too (before taking a screenshot). Since the data is refreshed when saving. And this is an asynchronous operation.